src/LeaseManager.cc

```diff
@@ -0,0 +1,307 @@
+/* Copyright (c) 2014-2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <string>
+
+#include "LeaseManager.h"
+
+#include "ExternalStorage.h"
+#include "LeaseCommon.h"
+#include "ShortMacros.h"
+
+namespace RAMCloud {
+
+/// Defines the number of leases the preallocator should try to keep ahead of
+/// the lastIssuedLeaseId.
```

Comment:
Say just a bit more to help readers understand why you chose this value. For example, in the worst case, this many leaseIds may be "lost" on a coordinator crash, so it needs to be small enough that we don't use up the 64-bit id space too quickly. On the other hand, we want it large enough to amortize the cost of a ZooKeeper operation, which can take several ms.